packages/inertia/src/inertia.js

```diff
@@ -136,11 +136,13 @@ export default {
     return response?.headers['x-inertia']
   },
 
-  cancelActiveVisits() {
-    if (this.cancelToken) {
-      this.cancelToken.cancel()
+  cancelActiveVisit({ cancelled = false, interrupted = false }) {
+    if (this.activeVisit) {
+      this.activeVisit.cancelToken.cancel()
+      this.fireEvent('finish', { detail: { visit: { ...this.activeVisit, completed: false, cancelled, interrupted } } } )
```

Comment:
Maybe we can place this in a generic onFinish method so that the event signature is only managed from a single place?

Reply:
So yeah, I am thinking about actually creating a series of event helpers:

```js
this.fireBeforeStartEvent()
this.fireErrorEvent()
this.fireFinishEvent()
this.fireInvalidEvent()
this.fireNavigateEvent()
this.fireProgressEvent()
this.fireStartEvent()
this.fireSuccessEvent()
```

Because it's becoming hard to keep track off, and is making the source code kind of messy.

Reply:
Could even place them in an `events.js` and then include that into the instance, to keep them separate from the rest 👍

Reply:
Yeah cool, let me try.

Reply:
And done 👍